keymanagementrequirements.md

```diff
@@ -0,0 +1,138 @@
+            Some regions may have high latency.
+        The certificates of TSAs MAY be revoked at any time without notices.
+            The removal of trust of VeriSign broke .NET 5+ NuGet. Thus Microsoft has to disable the package verification with a new release to unblock customers.
+
+Implications of not using a signed timestamp for a signature
+    In the absense of additional timestamp signature, the signature is only considered valid till key expiry. This may limit the use of short lived keys.
+    In case of key compromise it’s not possible to revoke signatures from a point in time where the time of compromise is known, as an attacker can create signatures with any signature time (by changing local time)
+
+### Signature Expiry
+
+Signatures can expire if
+
+    The certificate of the signing key or the timestamp key (whichever is longer) expires
+    The signed content indicates a expiry time
+
+In this section, the #2 scenario is focused.
+Scenarios
+
+The main scenario for using a configurable signature expiry is for the publisher to indicate how long they plan to maintain the signature on the artifact for. For example if a publisher signs an artifact with an expiry for 6 months, they are indicating that if within 6 months they identfy a reason that deployers should no longer trust the artifact they will rescind the signature.
+
+An added benefit is to defend against freeze attack. Since the signature expires in a short time although the signing key can live longer, the verifiers have to obtain the latest signature to verify, which implies that the verifies have access to the latest content.
+
+    Freeze Attack Similar to a replay attack, a freeze attack works by providing metadata that is not current. However, in a freeze attack, the attacker freezes the information a client sees at the current point in time to prevent the client from seeing updates, rather than providing the client older versions than the client has already seen. As with replay attacks, the attacker’s goal is ultimately to compromise a client who has vulnerable versions of packages installed. A freeze attack may be used to prevent updates in addition to having an installed package be out of date.
+
+### Transparent Root key auto rotation
+Root keys form the basis of hierarchical trust systems, where intermediate and leaf keys chain back to a root key, and leaf keys are used to generate signatures. Consumers of signed artifacts associate a baseline level of trust with signatures that chain to roots they trust. In the scenario that a root key is no longer trusted, all signed artifacts chaining back to the root are no longer trusted. The conditions which render the root keys to be no longer trusted, are disclosed to the customer through some other channel (CVE, internal security audit, etc.). Safe rotation of keys require a overlap period where both new and old root keys are trusted. This is always not possible, such as when a root key is compromised.
```

Comment:
> This is always **not** possible, such as when a root key is compromised.

This is not always possible, such as when a root key is compromised.

Reply:
Ack